modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/util/Utils.java

```diff
@@ -137,6 +141,33 @@ public static HttpResponse sendPOSTClaimMessage(HttpResponse response, String co
         return httpClient.execute(post);
     }
 
+    public static HttpResponse sendPOSTConsentMessage(HttpResponse response, String commonAuthUrl, String userAgent,
+                                                    String acsUrl, String artifact,  HttpClient httpClient, String
+                                                              pastreCookie) throws Exception {
+        String redirectUrl = getRedirectUrl(response);
+        Map<String, String> queryParams = getQueryParams(redirectUrl);
+
+
+        String sessionKey = queryParams.get("sessionDataKey");
+        String[] claims = queryParams.get("requestedClaims").split(",");
+
+        HttpPost post = new HttpPost(commonAuthUrl);
+        post.setHeader("User-Agent", userAgent);
+        post.addHeader("Referer", String.format(acsUrl, artifact));
+        post.addHeader("Cookie", pastreCookie);
+        List<NameValuePair> urlParameters = new ArrayList<>();
+
+        for (int i = 0; i < claims.length; i++) {
+            urlParameters.add(new BasicNameValuePair("consent_" + claims[i], "on"));
+        }
+        urlParameters.add(new BasicNameValuePair("sessionDataKey", sessionKey));
+        urlParameters.add(new BasicNameValuePair("requestedClaims", queryParams.get("requestedClaims")));
```

Comment:
We can remove `requestedClaims` and `mandatoryClaims` as they are not needed for the post handler.